client/app/components/packages/pas-form/edit.hbs

```diff
@@ -735,11 +743,16 @@
                 <div class="cell large-6 text-right">
                   <button
                     type="button"
-                    class="button large no-margin"
-                    {{on "click" (fn this.submit)}}
+                    class="button expanded large no-margin"
+                    disabled={{this.submit.isRunning}}
+                    {{on "click" (perform this.submit)}}
                     data-test-confirm-submit-button
                   >
-                    Submit
+                    {{#if this.submit.isRunning}}
+                      <FaIcon @icon="spinner" @spin="true" size="5x" class="light-gray" />
+                    {{else}}
+                      Submit
+                    {{/if}}
```

Comment:
Looks really good. I worry about this modal, I think it could be moved out to its own component.

Reply:
Yeah, i can see that  👍

Reply:
The modal already is its own component. Let's chat about what could be more isolated here. I'd love to improve the modal if we can.

Reply:
yup, okay! we can totally split it out more.